File: 00_Hackathons&Challenges/2018.10.16_GoogleTechDevGuide/src/calcInterpreter.java
public class calcInterpreter {

    public static void main(String[] args)
    {
        String[] cmds = {"+", "*"};
        int[] argus = {1, 3};

        System.out.println(interpret(1, cmds, argus));
    }

    public static int interpret(int value, String[] commands, int[] args) {
        int ans=value;

        for(int i =0; i< commands.length; i++){
            if(commands[i].equals("+")){
                ans = ans + args[i];
            }
            else if(commands[i].equals("-")){
                ans = ans - args[i];
            }
            else if(commands[i].equals("*")){
                ans = ans * args[i];
            }
            else{
                return -1;
            }
        }

        return ans;
    }
}

File: 00_Hackathons&Challenges/2018.10.18_GoogleTechDevGuidePython/calcinterpreter.py
def interpret(value, commands, args):
    ans = value
    #commands[i] obsolete just use i to refer to element at index i
    for cmd_val, args_val in zip(commands, args):
        if(cmd_val == '+'):
            ans = ans + args_val
        elif (cmd_val == '-'):
            ans = ans - args_val
        elif (cmd_val == '*'):
            ans = ans * args_val
        else:
            return -1
        # print(cmd_val, args_val, ans)

    return ans

cmds = ["+", "*"]
args = [1,3]

print(interpret(1, cmds, args))

# https://codingbat.com/prob/p234011